os/migrating-to-clcs.md

```diff
@@ -0,0 +1,377 @@
+
+```cloud-config
+#cloud-config
+
+ssh_authorized_keys:
+  - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+```
+
+In a Container Linux Config there is no analogous section to `ssh_authorized_keys`, but ssh keys for the core user can be set just as easily using the `passwd.users.*` section:
+
+```container-linux-config
+passwd:
+  users:
+    - name: core
+      ssh_authorized_keys:
+        - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+```
+
+### hostname
+
+In a cloud config the `hostname` section can be used to set a machine's hostname.
+
+```cloud-config
+#cloud-config
+
+hostname: "coreos1"
+```
+
+There is no analogous section in a Container Linux Config, so to set the hostname the `storage.files.*` section can be used to write `/etc/hostname`.
+
+```container-linux-config
+storage:
+  files:
+    - filesystem: "root"
+      path:       "/etc/hostname"
+      mode:       0644
+      contents:
+        inline: coreos1
+```
+
+### users
+
+The `users` section in a cloud config can be used to add users and specify many properties about them, from groups the user should be in to what the user's shell should be.
+
+```cloud-config
+#cloud-config
+
+users:
+  - name: "elroy"
+    passwd: "$6$5s2u6/jR$un0AvWnqilcgaNB3Mkxd5yYv6mTlWfOoCYHZmfi3LDKVltj.E8XNKEcwWm..."
+    groups:
+      - "sudo"
+      - "docker"
+    ssh-authorized-keys:
+      - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+```
+
+This same information can be added to the Container Linux Config in the `passwd.users.*` section.
+
+```container-linux-config
+passwd:
+  users:
+    - name:          "elroy"
+      password_hash: "$6$5s2u6/jR$un0AvWnqilcgaNB3Mkxd5yYv6mTlWfOoCYHZmfi3LDKVltj.E8XNKEcwWm..."
+      ssh_authorized_keys:
+        - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+      create:
+        groups:
+          - "sudo"
+          - "docker"
+```
+
+_Note: the `create:` section must exist or the user will not be created_
+
+### write_files
+
+The `write_files` section in a cloud config can be used to specify files and their contents that should be written to disk on the machine.
+
+```cloud-config
+#cloud-config
+write_files:
+  - path:        "/etc/resolv.conf"
+    permissions: "0644"
+    owner:       "root"
+    content: |
+      nameserver 8.8.8.8
+```
+
+This can be done in a Container Linux Config with the `storage.files.*` section.
+
+```container-linux-config
+storage:
+  files:
+    - filesystem: "root"
+      path:       "/etc/resolv.conf"
+      mode:       0644
+      contents:
+        inline: |
+          nameserver 8.8.8.8
+```
+
+When creating files in a Container Linux Config the filesystem the file is going to exist on must be referenced, and the provided path is relative to the root of the filesystem. This allows files to be written to filesystems other than the root filesystem.
```

Comment:
First sentence: "File specifications in this section of a CL Config must define the target filesystem and the file's path relative to the root of that filesystem."